### PR TITLE
176825410 discrete parameter values search

### PR DESCRIPTION
**Pull request by simonwa7**

### CODE REVIEW

File: src/python/zquantum/optimizers/search_points_optimizer.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        self.parameter_values_list = parameter_values_list
+
+    def _minimize(
+        self,
+        cost_function: CallableWithGradient,
+        initial_params: Optional[np.ndarray] = None,
+        keep_history: bool = False,
+    ) -> OptimizeResult:
+        """
+        Finds the parameters which minimize given cost function, by trying all the parameters from the grid.
```
Comment (mstechly) [inline src/python/zquantum/optimizers/search_points_optimizer.py:36]:
```suggestion
        Finds the parameters which minimize given cost function, by trying all the parameters from the provided list of points.
```

File: src/python/zquantum/optimizers/search_points_optimizer.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                min_value = value
+                optimal_params = parameter_values
+
+        return optimization_result(
+            opt_value=min_value,
+            opt_params=optimal_params,
+            nfev=len(self.parameter_values_list),
+            nit=None,
+            **construct_history_info(cost_function, keep_history)
+        )
```
Comment (mstechly) [inline src/python/zquantum/optimizers/search_points_optimizer.py:65]:
```suggestion
        )
        
```

File: src/python/zquantum/optimizers/search_points_optimizer.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        self,
+        parameter_values_list: List[np.ndarray],
+        recorder: RecorderFactory = _recorder,
+    ):
+        """
+        Args:
+            parameter_values_list: list of parameter values to evaluate
+            recorder: recorder object which defines how to store the optimization history.
+        """
+        super().__init__(recorder=recorder)
```
Comment (mstechly) [inline src/python/zquantum/optimizers/search_points_optimizer.py:26]: What about adding an assertion here that `len(parameter_values_list)>0`?
And appropriate test of course :)

**Review by mstechly — APPROVED** [review]

File: src/python/zquantum/optimizers/search_points_optimizer.py (hunk 3 above)
Comment (simonwa7) [reply src/python/zquantum/optimizers/search_points_optimizer.py:26]: Oh good point!

Comment (simonwa7) [reply src/python/zquantum/optimizers/search_points_optimizer.py:26]: Addressed here: d1556cd

**Pull request merged**